### ISSUE TITLE
[🐛 Bug]: DevTools not working with multiple windows and/or tabs.

### ISSUE THREAD

**Issue by AMP-BH** [SeleniumHQ/selenium#11436]
### What happened?

I was trying to test the block url with multiple windows or tab. But the urls are blocked only on the main window, not on newly opened tab or window.

### How can we reproduce the issue?

```shell
package com.seleniumtest.devtools;

import java.util.Optional;

import io.github.bonigarcia.wdm.WebDriverManager;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WindowType;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chromium.ChromiumDriver;
import org.openqa.selenium.devtools.DevTools;
import org.openqa.selenium.devtools.v107.network.Network;

import com.google.common.collect.ImmutableList;

public class DevToolBug {

  static DevTools devTools;
  static WebDriver driver = null;

  public static void main(String[] args) {

    try {
      WebDriverManager.chromedriver().setup();
      driver = new ChromeDriver();
      driver.manage().window().maximize();

      devTools = ((ChromiumDriver) driver).getDevTools();
      devTools.createSession();
      devTools.send(Network.enable(Optional.empty(), Optional.empty(), Optional.empty()));

      devTools.send(Network.setBlockedURLs(ImmutableList.of("*.css")));
      driver.get("https://amazon.com");

      driver.switchTo().newWindow(WindowType.TAB);
      devTools.send(Network.setBlockedURLs(ImmutableList.of("*.css")));
      driver.get("https://amazon.com");

      driver.switchTo().newWindow(WindowType.WINDOW);
      devTools.send(Network.setBlockedURLs(ImmutableList.of("*.css")));
      driver.get("https://amazon.com");
    } catch (Exception ex) {
      ex.printStackTrace();
    } finally {
      if (devTools != null) {
        devTools.close();
      }
      if (driver != null) {
        driver.quit();
      }
    }
  }
}
```

### Relevant log output

```shell
Task :DevToolBug.main()
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.
Starting ChromeDriver 108.0.5359.71 (1e0e3868ee06e91ad636a874420e3ca3ae3756ac-refs/branch-heads/5359@{#1016}) on port 42082
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
Dec 16, 2022 1:28:31 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
WARNING: Unable to find an exact match for CDP version 108, so returning the closest version found: 107
```

### Operating System

Windows 10

### Selenium version

4.6.0

### What are the browser(s) and version(s) where you see this issue?

108.0.5359.125 

### What are the browser driver(s) and version(s) where you see this issue?

Chrome Driver 107

### Are you using Selenium Grid?

NA

**Comment by AMP-BH** [comment]
@pujagani, both issues are different 11436 is about "How to use devTools across the windows and tabs?". and 11437 is about "If I create devtools session for each tab and window, how do i maintain them, as closing the devtools throws an exception?"

**Comment by AMP-BH** [comment]
@pujagani, in both the cases my usecase is to set user specified block urls for the new tab and window, and maintain the track of devtools sessions.

**Comment by diemol** [comment]
You have already solved this issue by the code you posted in #11437 
`devTools.createSession(driver.getWindowHandle());`

**Comment by AMP-BH** [comment]
Hello @diemol, the code in #11437, was just the work around. I was looking for more of a concrete solution, to provide utility in my framework code. 

But never the less will jump to next question then. THank you so much @diemol and @pujagani .
